tests/phpunit/CRM/MembersOnlyEvent/BAO/EventGroupTest.php

```diff
@@ -51,4 +53,89 @@ public function testGetAllowedGroupIDs() {
 
   }
 
+  /**
+   * Tests getContactAllowedGroups().
+   */
+  public function testIfGetContactAllowedGroupsReturnsNothingIfNoAllowedGroupsAreConfigured() {
+    $params = [
+      'first_name' => 'John',
+      'last_name' => 'Doe',
+    ];
+    $contactID = ContactFabricator::fabricate($params)['id'];
+    $membersOnlyEvent = MembersOnlyEventFabricator::fabricate();
+    $studentsGroup = GroupFabricator::fabricate(['title' => 'Students'], TRUE);
+
+    $params = [
+      'contact_id' => $contactID,
+      'group_id' => $studentsGroup->id,
+      'status' => 'Added',
+    ];
+    $contactGroups = civicrm_api3('GroupContact', 'create', $params);
+
+    // Tests the contact does not belong to any allowed groups.
```

Comment:
no need for this, the test method name already explains what the test does, same for all the inline comments below.